scale/job/management/commands/scale_pre_steps.py

```diff
@@ -95,28 +95,25 @@ def _get_job_exe(self, job_id, exe_num):
     def _generate_input_metadata(self, job_exe):
         """Generate the input metadata file for the job execution
 
-        :param job_id: The job ID
-        :type job_id: int
-        :param exe_num: The execution number
-        :type exe_num: int
+        :param job_exe: The job_exe model
+        :type job_exe: `job.models.JobExecution`
         """
 
         job_interface = job_exe.job_type.get_job_interface()
-
         if not job_interface.needs_input_metadata():
             return
 
         # Generate input metadata dict
         input_metadata = {}
-        config = job_exe.get_execution_configuration
+        config = job_exe.get_execution_configuration()
         if 'input_files' in config.get_dict():
             input_metadata['JOB'] = {}
             input_data = job_exe.job.get_input_data()
             for i in input_data.values.keys():
                 if type(input_data.values[i]) is JsonValue:
                     input_metadata['JOB'][i] = input_data.values[i].value
                 elif type(input_data.values[i]) is FileValue:
-                    input_metadata['JOB'][i] = [ScaleFile.objects.get(pk=f)._get_url() for f in
+                    input_metadata['JOB'][i] = [ScaleFile.objects.get_details(file_id=f).__dict__ for f in
```

Comment:
Use a serializer here? Might be hard to read/write python objects from the file.